libs/verify-ios-ci-environment.sh

```diff
@@ -22,11 +22,6 @@ rustup target add "${RUST_TARGETS[@]}"
 
 # If you add a dependency below, mention it in building.md in the iOS section!
 
-if ! [[ -x "$(command -v carthage)" ]]; then
-  echo 'Error: Carthage needs to be installed. See https://github.com/Carthage/Carthage#installing-carthage for install instructions.' >&2
-  exit 1
-fi
-
 if ! [[ -x "$(command -v protoc-gen-swift)" ]]; then
```

Comment:
nit: we don't have to do this now, but we no longer depend on anything protobuf swift related so we can probably get rid of this too 😄

Reply:
good idea! I'll probably keep that separate since idk how viaduct interacts with protobuf or what that story is (I _think_ it's fine but just to be safe).